Celeste.Mod.mm/Mod/Helpers/TypeHelper.cs

```diff
@@ -213,37 +208,85 @@ public static class TypeHelper {
             };
         }
 
-        public static void Add_DataNameTypeMap(string name, Type type, bool overwrite = false) {
-            if (overwrite || !entityDataNameToType.ContainsKey(name))
-                entityDataNameToType[name] = type;
+        public readonly static Dictionary<string, Type> FullName_to_Type;
+
+        public readonly static Dictionary<string, Type> EntityDataName_to_Type;
```

Comment:
Using underscores in the middle of names is not the C# naming scheme. `FullNameToType` would be better. (same goes for all methods in this cass)

I still don't think this needs to be public, especially not publicly mutable - `CheckEntityByDataName` and `Link_DataName_To_Type` exists already for `EntityDataName_to_Type`, `FullName_to_Type` is just a cache for `TryGetType`.
Only publicly exposing a `IReadOnlyDictionary` view for `EntityDataName_to_Type` would be much more reasonable (albeit only useful if you need to enumerate through it, for whatever reason) and would leave more space for refactoring later. `FullName_to_Type` being public seems useless, imo.

Reply:
I agree with the nomenclature fix, but I was initially trying to make it more legible, since some words like EntityData use capitals anyways which makes it a bit more annoying to parse for non-English speakers.

For the `private` vs public exposure of an IReadOnlyDictionary, this doesn't resolve the internal issue since with an IReadOnlyDictionary you can't lock the values of the IReadOnlyDictionary, you just can't add/remove keys from it, if I'm reading your suggestion right. An ImmutableDictionary *is* a more viable option for what you're asking for, although that practically requires a deepcopy to construct/pass so that's really bad and slow, which is why I went with it being public. I'll swap it over to a private Dictionary and then a public IReadOnlyDictionary as a getter